pkg/corerp/api/v20231001preview/container_conversion.go

```diff
@@ -151,6 +157,65 @@ func (src *ContainerResource) ConvertTo() (v1.DataModelInterface, error) {
 	return converted, nil
 }
 
+// toEnvDataModel: Converts from versioned datamodel to base datamodel
+func toEnvironmentVariableDataModel(e map[string]*EnvironmentVariable) (map[string]datamodel.EnvironmentVariable, error) {
+
+	m := map[string]datamodel.EnvironmentVariable{}
+
+	for key, val := range e {
+		if val == nil {
+			return nil, v1.NewClientErrInvalidRequest(fmt.Sprintf("Environment variable %s is nil", key))
+		}
+		// An environment variable can have either value(Value) or secret value(ValueFrom), but not both
+		if val.Value != nil && val.ValueFrom != nil {
+			return nil, v1.NewClientErrInvalidRequest(fmt.Sprintf("Environment variable %s has both value and secret value", key))
+		}
+
+		if val.Value != nil {
+			m[key] = datamodel.EnvironmentVariable{
+				Value: val.Value,
+			}
+		} else {
+			m[key] = datamodel.EnvironmentVariable{
+				ValueFrom: &datamodel.EnvironmentVariableReference{
+					SecretRef: &datamodel.EnvironmentVariableSecretReference{
+						Source: to.String(val.ValueFrom.SecretRef.Source),
+						Key:    to.String(val.ValueFrom.SecretRef.Key),
+					},
+				},
+			}
+
+		}
+
+	}
+	return m, nil
+}
+
+// fromEnvDataModel: Converts from base datamodel to versioned datamodel
```

Comment:
nit: I think we may want to keep the same name format. If it's  toEnvironmentVariableDataModel() then a corresponding fromEnvironmentVariableDataModel()